ReactQt/runtime/src/reactuimanager.cpp

```diff
@@ -128,8 +128,8 @@ void ReactUIManager::setChildren
 {
   //TODO: This is a simple implementation which fixes a broken example. It's not properly tested and may need revisiting
   QList<int> indices;
-  foreach(int i, childrenTags) {
-    indices.append(0); //0 is a default z-order in qml
+  for(int i = 0; i < childrenTags.size(); ++i) {
+    indices.append(i); //Incrementing per each child. Similar to iOS. See the comment in manageChildren
```

Comment:
@MaxRis , of course I'm gonna take a closer look on this, but I'm very curious about your solution (maybe I'm missing something). 
When I comment out setting Z order for the children and passing all 0 indices, I have same situation with text - it's drown at the bottom.